### PR TITLE
Tests for AnimationTimeline interface

### PR DESCRIPTION
**Pull request by alsemenov**
Tests for AnimationTimeline interface

### CODE REVIEW

**Comment by sgrekhov** [comment]
newAnimation() helper function has been removed

File: web-animations/animation-timeline/animation-timeline-play.html [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    });
+}, 'Test AnimationTimeline.play() method accociates returned AnimationPlayer ' +
+    'with timeline instance used to call method play(). Using timeline of another document');
+
+async_test(function(t) {
+    var player = document.timeline.play(new Animation(createDiv(t), KEYFRAMES, ANIMATION_DURATION));
+
+    player.ready.then(t.step_func(function() {
+        setTimeout(t.step_func(function() {
+            assert_approx_equals(player.currentTime, 50, TIME_EPSILON, 'AnimationTimeline.play() ' +
```
Comment (alancutter) [inline web-animations/animation-timeline/animation-timeline-play.html]: Mixing the ANIMATION_DURATION constant with magic numbers like 50 forces the reader to look up ANIMATION_DURATION in testcommon.js. The use of magic numbers or constants should be consistent without mixing them together.
Please resolve this here and elsewhere in this change.

Comment (sgrekhov) [inline web-animations/animation-timeline/animation-timeline-play.html]: Magic numbers replaced by constants

File: web-animations/testcommon.js [hunk 2]
```diff
@@ -56,3 +61,12 @@ function removeElement(element) {
...
     return Object.prototype.toString.call(object).slice(8, -1);
 }
+
+// Returns expected top of the target element at currentTime
+function getExpectedTop(currentTime, endTime) {
+    if (!endTime) {
+        endTime = ANIMATION_DURATION;
+    }
+    return ANIMATION_TOP_0 + (ANIMATION_TOP_1 - ANIMATION_TOP_0)
+        * (currentTime % endTime);
```
Comment (ericwilligers) [inline web-animations/testcommon.js]: currentTime / endTime

Comment (sgrekhov) [inline web-animations/testcommon.js]: I think currentTime % endTime is Ok here. Example. ANIMATION_TOP_0 = 0, ANIMATION_TOP_1 = 100, endTime = 1000, currentTime = 1100. At 1100 expected value is 10, not 110. That's how tests assumes.  For the case when iterationComposite = 'accumulate' we use code like 
            assert_approx_equals(window.getComputedStyle(target).top,
                N * ANIMATION_TOP_1 + getExpectedTop(...), COMPUTED_STYLE_EPSILON,  '...');
;

Comment (alancutter) [inline web-animations/testcommon.js]: It's not very clear you intend this to test expected iteration values, adding a line like `var iterationFraction = (currentTime / iterationDuration) % 1;` would make this more clear.
Should endTime be renamed iterationDuration? In the spec "end time" refers to the time at which a node finishes its active phase with respect to its parent's timeline which is quite different to how it's being used here.
Also KEYFRAMES doesn't define a linear animation, you'll need to incorporate ANIMATION_TOP_0_5 here as well.

**Comment by alancutter** [comment]
lgtm after these last two comments are resolved, sorry for such long review cycles.

**Comment by sgrekhov** [comment]
Both of the issues mentioned by Alan are resolved. Please review and merge to w3c if everithing is Ok

**Comment by ericwilligers** [comment]
This is fine, please resolve merge conflicts, and ask Ms2ger to merge.
(Alan and I don't have permission to merge.)

**Comment by birtles** [comment]
I'm going to close this unmerged because:
* I don't think we want to add tests for level 2 features yet since that is unstable
* The tests are no up-to-date (e.g. they include tests for `timeline.play()` and `timeline.getAnimationPlayers()` which no longer exist)
* We have tests for timeline under web-animations/animation/finished.html
* I don't have time to go through and see if there is anything salvageable here (from a glance there isn't, except perhaps the idlharness tests which we will likely add for all interfaces at once) and the submitter is no longer available to rebase and incorporate any review feedback if I did

**Pull request closed without merging**